File: CSharp/Concentus/Silk/Structs/NLSFCodebook.cs
using Concentus.Common.CPlusPlus;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concentus.Silk.Structs
{
    /// <summary>
    /// Structure containing NLSF codebook
    /// </summary>
    internal class NLSFCodebook
    {
        internal short nVectors = 0;

        internal short order = 0;

        /// <summary>
        /// Quantization step size
        /// </summary>
        internal short quantStepSize_Q16 = 0;

        /// <summary>
        /// Inverse quantization step size
        /// </summary>
        internal short invQuantStepSize_Q6 = 0;

        /// <summary>
        /// POINTER
        /// </summary>
        internal Pointer<byte> CB1_NLSF_Q8 = null;

        /// <summary>
        /// POINTER
        /// </summary>
        internal Pointer<byte> CB1_iCDF = null;

        /// <summary>
        /// POINTER to Backward predictor coefs [ order ]
        /// </summary>
        internal Pointer<byte> pred_Q8 = null;

        /// <summary>
        /// POINTER to Indices to entropy coding tables [ order ]
        /// </summary>
        internal Pointer<byte> ec_sel = null;

        /// <summary>
        /// POINTER
        /// </summary>
        internal Pointer<byte> ec_iCDF = null;

        /// <summary>
        /// POINTER
        /// </summary>
        internal Pointer<byte> ec_Rates_Q5 = null;

        /// <summary>
        /// POINTER
        /// </summary>
        internal Pointer<short> deltaMin_Q15 = null;
        
        internal void Reset()
        {
            nVectors = 0;
            order = 0;
            quantStepSize_Q16 = 0;
            invQuantStepSize_Q6 = 0;
            CB1_NLSF_Q8 = null;
            CB1_iCDF = null;
            pred_Q8 = null;
            ec_sel = null;
            ec_iCDF = null;
            ec_Rates_Q5 = null;
            deltaMin_Q15 = null;
        }
    }
}

File: go/opus/NLSFCodebook.go
package opus

type NLSFCodebook struct {
	nVectors            int16
	order               int16
	quantStepSize_Q16   int16
	invQuantStepSize_Q6 int16
	CB1_NLSF_Q8         []int16
	CB1_iCDF            []int16
	pred_Q8             []int16
	ec_sel              []int16
	ec_iCDF             []int16
	ec_Rates_Q5         []int16
	deltaMin_Q15        []int16
}

func (this *NLSFCodebook) Reset() {
	this.nVectors = 0
	this.order = 0
	this.quantStepSize_Q16 = 0
	this.invQuantStepSize_Q6 = 0
	this.CB1_NLSF_Q8 = nil
	this.CB1_iCDF = nil
	this.pred_Q8 = nil
	this.ec_sel = nil
	this.ec_iCDF = nil
	this.ec_Rates_Q5 = nil
	this.deltaMin_Q15 = nil
}
